src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts

```diff
@@ -0,0 +1,72 @@
+import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import 'rxjs/add/observable/forkJoin';
+import 'rxjs/add/observable/of';
+import { Observable } from 'rxjs/Observable';
+
+import * as _ from 'lodash';
+
+@Injectable()
+export class RgwBucketService {
+
+  private url = '/api/rgw/proxy/bucket';
+
+  constructor(private http: HttpClient) { }
+
+  list() {
+    return this.http.get(this.url)
+      .flatMap((buckets: string[]) => {
+        if (buckets.length > 0) {
+          return Observable.forkJoin(
+            buckets.map((bucket: string) => {
+              return this.get(bucket);
+            })
+          );
+        }
+        return Observable.of([]);
+      });
+  }
+
+  get(bucket: string) {
+    let params = new HttpParams();
+    params = params.append('bucket', bucket);
+    return this.http.get(this.url, { params: params });
+  }
+
+  create(bucket: string, uid: string) {
+    const body = JSON.stringify({
+      'bucket': bucket,
+      'uid': uid
+    });
+    return this.http.post(`/api/rgw/bucket`, body);
+  }
+
+  update(bucketId: string, bucket: string, uid: string) {
+    let params = new HttpParams();
+    params = params.append('bucket', bucket);
+    params = params.append('bucket-id', bucketId as string);
+    params = params.append('uid', uid);
+    return this.http.put(this.url, null, { params: params });
```

Comment:
Shouldn't we send the data in the body of the request? and just append the id of the bucket to the url?

Reply:
This method has been removed because it is not required in the read-only UI. To answer your question, no we need to submit the args this way because the RGW Admin OPS API want to see them this way.